apps/daffio/src/app/core/sidebar/effects/sidebar.effects.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { TestBed } from '@angular/core/testing';
+
+import { Observable } from 'rxjs';
+
+import { Action } from '@ngrx/store';
+import { ROUTER_NAVIGATION } from '@ngrx/router-store';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { Actions } from '@ngrx/effects';
+
+import { hot, cold, getTestScheduler } from 'jasmine-marbles';
+
+import { SidebarEffects } from './sidebar.effects';
+import * as SidebarActions from '../actions/sidebar.actions';
+
+describe('Daffio | SidebarEffects', () => {
+  let effects: SidebarEffects;
+  let actions$: Observable<any>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        SidebarEffects,
+        provideMockActions(() => actions$)
+      ]
+    });
+
+    effects = TestBed.get(SidebarEffects);
+  });
+
+  it('should be created', () => {
+    expect(effects).toBeTruthy();
+  });
+
+  describe('when a ROUTER_NAVIGATION occurs', () => {
```

Comment:
This test doesn't really test for "when a ROUTER_NAVIGATION occurs", right? It looks like it tests for "when closeOnPageChange$ is triggered". The `action` constant below doesn't seem to be used anywhere. If Jasmine marbles is being too annoying, I'm ok with just testing the latter, but I don't think this test truly tests an effect triggered by routing navigation

Reply:
Oh, my bad. I misunderstood how `effects.closeOnPageChange$(10, getTestScheduler())` worked. I thought it was manually calling the effect, but I guess it's actually more like `effects.closeOnPage...` is functioning like a watch or spy.